nanodbc/nanodbc.h

```diff
@@ -1282,6 +1285,69 @@ class statement
 /// \brief Manages and encapsulates ODBC resources such as the connection and environment handles.
 class connection
 {
+
+private:
+    class connection_impl;
+    friend class nanodbc::transaction::transaction_impl;
+
+#if __cpp_lib_variant >= 201606L
+public:
+    /// \brief A class representing a connection attribute.
+    ///
+    /// Callers should create attributes using the 3 argument constructor.
+    /// First argument is the Attribute argument to the ODBC SQLSetConnectAttr
+    /// function.  The second is the StringLength, and the third is used to
+    /// inform the ValuePtr argument to SQLSetConnectAttr.  This argument,
+    /// a std::variant, is a type safe union of std::vector<uint8_t> ( binary
+    /// buffer payloads ), nanodbc::string ( string payloads ), or std::(u)intptr_t,
+    /// for both u/int payloads, as well as pointers to more generic buffers.
+    ///
+    /// See https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlsetconnectattr-function
+    class attribute
+    {
+    public:
+        attribute() = delete;
+        attribute& operator=(attribute const&) = delete;
+        attribute(attribute const& other);
+        attribute(
+            long const& attr,
+            long const& strLen,
+            std::variant<std::vector<uint8_t>, string, std::intptr_t, std::uintptr_t> const& rsrc);
+
+    private:
+        void extractValuePtr();
+
+        long odbcAttribute;
+        long odbcStringLength;
+        std::variant<std::vector<uint8_t>, string, std::intptr_t, std::uintptr_t> resource;
+        void* odbcValuePtr;
```

Comment:
First, please don't use `camelCase` please.

The `odbc` seems redundant. 
I'd go for `odbcAttribute` -> `attribute_` where `_` distinguishes the member from `attribute` class and `odbcValuePtr` > `value_ptr`, etc.